frontend/js/src/playlists/UserSearch.tsx

```diff
@@ -0,0 +1,99 @@
+import * as React from "react";
+import { faUser } from "@fortawesome/free-solid-svg-icons";
+import GlobalAppContext from "../utils/GlobalAppContext";
+import ListenControl from "../listens/ListenControl";
+
+export type UserSearchProps = {
+  userClick: (event: string) => void;
+  placeholder: string;
+};
+
+export type UserSearchState = {
+  newUser: string;
+  userSearchResults: Array<any>;
+};
+
+export default class UserSearch extends React.Component<
+  UserSearchProps,
+  UserSearchState
+> {
+  static contextType = GlobalAppContext;
+  declare context: React.ContextType<typeof GlobalAppContext>;
+
+  constructor(props: UserSearchProps) {
+    super(props);
+    this.state = {
+      newUser: "",
+      userSearchResults: [],
+    };
+  }
+
+  componentDidUpdate(prevProps: UserSearchProps, prevState: UserSearchState) {
+    const { newUser } = this.state;
+
+    if (prevState.newUser !== newUser) {
+      this.searchUsers();
+    }
+  }
+
+  handleInputChange = (
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
+    const { target } = event;
+    const value =
+      target.type === "checkbox"
+        ? (target as HTMLInputElement).checked
+        : target.value;
+    const { name } = target;
+    // @ts-ignore
+    this.setState({
+      [name]: value,
+    });
+  };
```

Comment:
I see this is copied from another component, where multiple inputs and checkboxes use the same function.
In the case of this component, this handler can be rewritten like so:
```suggestion
  handleInputChange = (
    event: React.ChangeEvent<HTMLInputElement>
  ) => {
    this.setState({
      newUser: event.target.value
    });
  };
```

With that being said, we probably want to enhance this method so as not to call it on every keyboard stroke, which could slow down the application. We can use the throttle utility function from the lodash library (already installed) like it is done here:
https://github.com/metabrainz/listenbrainz-server/blob/e04252cc5b5213853991528de6254486f51b586c/frontend/js/src/brainzplayer/BrainzPlayer.tsx#L610-L613

In short: for any method `myFunction` on your class, define a new `throttledMyFunction` which will be the output of calling [lodahs's throttle](https://lodash.com/docs/4.17.15#throttle) with `myFunction` as the first argument and a number of milliseconds to wait between calls as the second argument (let's set this delay at 300ms).
Then your input will take `onChange={this.throttledMyFunction}`, or in our case `throttledHandleInputChange`.